geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/clustering/ClusteringUtils.java

```diff
@@ -0,0 +1,409 @@
+import mil.nga.giat.geowave.index.sfc.data.NumericData;
+import mil.nga.giat.geowave.index.sfc.data.NumericRange;
+import mil.nga.giat.geowave.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.store.index.CustomIdIndex;
+import mil.nga.giat.geowave.store.index.DimensionalityType;
+import mil.nga.giat.geowave.store.index.Index;
+import mil.nga.giat.geowave.store.index.IndexType;
+import mil.nga.giat.geowave.store.query.DistributableQuery;
+import mil.nga.giat.geowave.store.query.SpatialQuery;
+import mil.nga.giat.geowave.vector.adapter.FeatureDataAdapter;
+
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.iterators.user.RegExFilter;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LinearRing;
+import com.vividsolutions.jts.geom.Polygon;
+import com.vividsolutions.jts.geom.PrecisionModel;
+import com.vividsolutions.jts.geom.impl.CoordinateArraySequence;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+
+public class ClusteringUtils
+{
+
+	final static Logger LOGGER = LoggerFactory.getLogger(ClusteringUtils.class);
+	private static BasicAccumuloOperationsBuilder basicAccumuloOperationsBuilder = new DirectBasicAccumuloOperationsBuilder();
+
+	public static void setBasicAccumuloOperationsBuilder(
+			BasicAccumuloOperationsBuilder basicAccumuloOperationsBuilder ) {
+		ClusteringUtils.basicAccumuloOperationsBuilder = basicAccumuloOperationsBuilder;
+	}
+
+	/*
+	 * null field will result in all possible entries in that field
+	 */
+	public static IteratorSetting createScanIterator(
+			final String iterName,
+			final String rowRegex,
+			final String colfRegex,
+			final String colqRegex,
+			final String valRegex,
+			final boolean orFields ) {
+		final IteratorSetting iter = new IteratorSetting(
+				15,
+				iterName,
+				RegExFilter.class);
+		RegExFilter.setRegexs(
+				iter,
+				rowRegex,
+				colfRegex,
+				colqRegex,
+				valRegex,
+				orFields);
+
+		return iter;
+	}
+
+	public static BasicAccumuloOperations createOperations(
+			final String zookeeper,
+			final String accumuloInstance,
+			final String accumuloUser,
+			final String accumuloPassword,
+			final String namespace )
+			throws Exception {
+
+		return basicAccumuloOperationsBuilder.build(
+				zookeeper,
+				accumuloInstance,
+				accumuloUser,
+				accumuloPassword,
+				namespace);
+	}
+
+	/*
+	 * generate a polygon with 4 points defining a boundary of lon: -180 to 180
+	 * and lat: -90 to 90 in a counter-clockwise path
+	 */
+	public static Polygon generateWorldPolygon() {
```

Comment:
it seems this is just used by tests, in general I'd rather not even provide this method to tempt a developer to use it for a query. I remember Binh had it as the query when he intended to do a full table scan - we perhaps should have logic deeper in the bowels of GeoWave to identify queries that extend fully across the range in any one dimension and map that to a full table scan, but until then, we do query decomposition on this, and map it to accumulo ranges which is just less efficient.  Like I said, this isn't actually used by anything other than test, but I think its better to just remove it so its not used when a developer is looking to do a full table scan.